### PR TITLE
[terra-clinical-header]Accessibility Updates

### LINKED ISSUE #791: [terra-clinical-header] Accessibility Updates
# Feature Request

## Description
As a consumer of Terra-Clinical-Header, I would like to the user to be able to select a heading level, so that it does not have accessibility assumptions or defaults that create incorrect context for users and my application meets [WCAG Success Criterion 1.3.1 (Info and Relationships)](https://www.w3.org/TR/WCAG21/#info-and-relationships) plus Success Criterion 1.3.2 (Meaningful Sequence).
When headings [reflect page organization](https://www.w3.org/WAI/tutorials/page-structure/headings/#headings-that-reflect-the-page-organization), they should be given ‘id’s and used with ‘aria-labelledby’ to associate headings with the corresponding page region​
example: https://www.w3.org/WAI/tutorials/page-structure/example/

### PR DESCRIPTION
**Pull request by saket2403**
### Summary

Allow users to be able to select a heading level, so that it does not have accessibility assumptions or defaults that create incorrect context for users and my application meets [WCAG Success Criterion 1.3.1 (Info and Relationships)](https://www.w3.org/TR/WCAG21/#info-and-relationships) plus Success Criterion 1.3.2 (Meaningful Sequence).
Allow users to be able to add id to the header to be used with ‘aria-labelledby’ to associate headings with the corresponding page region​

Closes #791 

### Deployment Link

[Link](https://terra-clinic-clinical-h-n4d67b.herokuapp.com/components/terra-clinical-header/clinical-header/clinical-header)

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by xenoworf — APPROVED** [review]

**Comment by xenoworf** [comment]
I added the major version bump label to this PR since I saw there are breaking changes.

File: packages/terra-clinical-header/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+* Breaking Changes
+  * Changed `title` prop to `text` prop.
+  * Added `level` prop to allow users to render different heading levels.
+  * Added `id` prop.
```
Comment (benbcai) [inline packages/terra-clinical-header/CHANGELOG.md:8]: Can these be updated to be more descriptive like how they are in the upgrade guide below? When consumers look at these, they know what was changed but hard to understand why the change is made and how it would impact them.

File: packages/terra-clinical-header/src/Header.jsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ const propTypes = {
   /**
    * Text to be displayed as the title in the header bar
    */
-  title: PropTypes.string,
+  text: PropTypes.string,
```
Comment (benbcai) [inline packages/terra-clinical-header/src/Header.jsx:26]: Can this and the other new props be alphabetized for consistency?

File: packages/terra-clinical-header/CHANGELOG.md (hunk 1 above)
Comment (saket2403) [reply packages/terra-clinical-header/CHANGELOG.md]: Updated [53ced76](https://github.com/cerner/terra-clinical/pull/792/commits/53ced7677e4fd6eb2b3bea9465b779d456355f14)

File: packages/terra-clinical-header/src/Header.jsx (hunk 2 above)
Comment (saket2403) [reply packages/terra-clinical-header/src/Header.jsx]: Arranged [53ced76](https://github.com/cerner/terra-clinical/pull/792/commits/53ced7677e4fd6eb2b3bea9465b779d456355f14)

File: packages/terra-clinical-header/src/terra-dev-site/doc/clinical-header/AccessibilityGuide.2.doc.mdx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { Badge } from 'terra-clinical-header/package.json?dev-site-package';
+
+<Badge />
+
+# Accessibility Guide for Terra Clinical Header
+
+Clinical Header allows users to organise web pages using the six levels of section heading. Where `level={1} (<h1>)` is the highest and `level={6} (<h6>)` is the lowest heading level.
```
Comment (benbcai) [inline packages/terra-clinical-header/src/terra-dev-site/doc/clinical-header/AccessibilityGuide.2.doc.mdx:7]:
```suggestion
Clinical Header allows users to organize web pages using the six levels of section heading with `level={1} (<h1>)` being the highest and `level={6} (<h6>)` being the lowest heading level.
```

**Review by benbcai — APPROVED** [review]

File: packages/terra-clinical-header/src/Header.jsx [hunk 4]
```diff
@@ -16,46 +16,61 @@ const propTypes = {
...
-   * Text to be displayed as the title in the header bar
+   * A Boolean indicating if element is a subheader.
    */
-  title: PropTypes.string,
+  isSubheader: PropTypes.bool,
 
   /**
-   * Content to be displayed at the end of the header
-   * The element passed as endContent will be wrapped in a div with flex attributes.
+   * Sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. This helps screen readers to announce appropriate heading levels.
```
Comment (sdadn) [inline packages/terra-clinical-header/src/Header.jsx:35]:
```suggestion
   * Sets the heading level to `1`, `2`, `3`, `4`, `5` or `6`. This helps screen readers to announce appropriate heading levels.
```

File: packages/terra-clinical-header/src/Header.jsx [hunk 5]
```diff
@@ -16,46 +16,61 @@ const propTypes = {
...
+   * Changing 'level' will not visually change the style of the content.
    */
-  endContent: PropTypes.element,
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
```
Comment (ShettyAkarsh) [inline packages/terra-clinical-header/src/Header.jsx:38]: i am not sure, does this values more sense for consumers. if we mention it as strings ? "One", "two" etc

cc. @neilpfeiffer

**Comment by sdadn** [comment]
Our team's stance on PRs is that every commit merged on `main` should be releasable. Since this is an active change resulting in a major version bump and we are not ready to do a major version release yet,  I updated the base branch to a new feature branch `A11y_MVB`.

**Review by supreethmr — APPROVED** [review]

File: packages/terra-clinical-header/src/Header.jsx (hunk 5 above)
Comment (benbcai) [reply packages/terra-clinical-header/src/Header.jsx:38]: I think using number is ok because they really are `h1`, `h2`, etc. If we used strings we would have to map them back to a number to be able to do this:

> const HeaderElement = `h${level}`;

**Review by ShettyAkarsh — APPROVED** [review]
